samcli/lib/build/build_strategy.py

```diff
@@ -111,7 +111,8 @@ def build_single_function_definition(self, build_definition):
                              single_build_dir,
                              build_definition.metadata
         )
-        function_build_results[single_function_name] = result
+        for function in build_definition.functions:
+            function_build_results[function.name] = result
```

Comment:
Looking at the merge conflict, it seems this change might break the change made in this [PR](https://github.com/aws/aws-sam-cli/pull/2623).